### PR TITLE
feat(colors, highlights): improved `NonText` readability

### PR DESCRIPTION
**Pull request by AlejandroSuero**
@scottmckendry a little tweak on the colours and the highlights to improve readability when using `NonText` and `NonText+CursorLine`. Usually used by AI plugins and terminal based plugins to show "ghost text".

> [!WARNING]
> Toggling light and dark 💥

- Before:

https://github.com/scottmckendry/cyberdream.nvim/assets/71392160/db0515f1-6054-449e-8bd8-a3374c18b8e4

Using `:term`:

https://github.com/scottmckendry/cyberdream.nvim/assets/71392160/4cdc091e-6cc2-41c6-b30c-8c9a60d1f20c

- After:

https://github.com/scottmckendry/cyberdream.nvim/assets/71392160/0d5aba57-1b0d-4bbb-883b-e1727859a412

Using `:term`:

Screenshot 2024-06-15 at 00 16 14

### CODE REVIEW

**Comment by AlejandroSuero** [comment]
I force pushed cause I don't know how (87aee45) got in the PR while I was writing it 🤔

**Comment by scottmckendry** [comment]
Wow @AlejandroSuero ! Appreciate all the work you're putting in on these PRs!

I think this is a very necessary change and should be fixed. However, I think we should try and make it work without affecting the base palette.  As you can probably already tell, this has the knock-on effect of updating _all_ of the extras that reference it.

You can use the `util.blend` function to produce new colors from the ones already available if there is nothing suitable in the palette itself.

**Comment by AlejandroSuero** [comment]
@scottmckendry okay, I will play around to get a feel to mix the base palette.

File: lua/cyberdream/colors.lua
Comment (scottmckendry) [inline lua/cyberdream/colors.lua:1]: avoid updating the base palette if possible :)

**Review by scottmckendry — CHANGES_REQUESTED** [review]

File: lua/cyberdream/colors.lua
Comment (AlejandroSuero) [reply lua/cyberdream/colors.lua:1]: @scottmckendry I made the changes in (1eea635)

https://github.com/scottmckendry/cyberdream.nvim/assets/71392160/4da406e3-68be-4a8a-968d-55d32e950625

I made it so it differentiates from the `Comment` highlight so it doesn't conflict.

**Review by scottmckendry — CHANGES_REQUESTED** [review]
It's still showing lots of changes to extras, I think there still may be changes in `colors.lua` that differ from `main`. Can you check, please? Thanks!

**Comment by AlejandroSuero** [comment]
@scottmckendry This has nothing to do with this PR but while I was playing around with the values, I used Telescope a lot and notice that there is no contrast between search and results and preview.

Here is my proposal.

- Without changing anything but `TelescopePrompt{Border,Normal,Prefix}`):

Changed with red theme

- Changing the colours:

Changed with a blueish theme

**Comment by AlejandroSuero** [comment]
> It's still showing lots of changes to extras, I think there still may be changes in `colors.lua` that differ from `main`. Can you check, please? Thanks!

Okay I am going to copy the main branch colors.

**Comment by AlejandroSuero** [comment]
@scottmckendry now it only changes 1 files, I check the colour contrast and still works correctly both in light and dark

**Review by scottmckendry — APPROVED** [review]

**Comment by scottmckendry** [comment]
> @scottmckendry This has nothing to do with this PR but while I was playing around with the values, I used Telescope a lot and notice that there is no contrast between search and results and preview.
> 
> Here is my proposal.
> 
> * Without changing anything but `TelescopePrompt{Border,Normal,Prefix}`):
> 
> Changed with red theme
> * Changing the colours:
> 
> Changed with a blueish theme

@AlejandroSuero the blue looks quite nice. Do you have this change in a branch? Would like to see how it looks in person :)

**Pull request merged**